### PR TITLE
Feature/response building

### PR DESCRIPTION
**Pull request by Swess**

### CODE REVIEW

File: src/HttpLib/HttpRequestHandler.java [hunk 1]
```diff
@@ -43,7 +44,12 @@ public HttpResponse send(HttpRequest request) throws InvalidRequestException, IO
         in.close();
         socket.close();
 
-        return new HttpResponse(res.toString());
+        try {
+            return new HttpResponse(res.toString());
+        } catch (HttpFormatException e) {
+            // Invalid response format received, sending invalid empty reponse
+            return new HttpResponse();
```
Comment (Nwitz) [inline src/HttpLib/HttpRequestHandler.java]: Some sort of error should be thrown here, catching exceptions and doing nothing with them will lead to harder debugging.

File: src/HttpLib/HttpRequestMethod.java [hunk 2]
```diff
@@ -15,4 +25,8 @@
     public String toString() {
         return _val;
     }
+
+    public static HttpRequestMethod get(String methodString) {
+        return lookup.get(methodString);
```
Comment (Nwitz) [inline src/HttpLib/HttpRequestMethod.java:30]: Was this tested with invalid values?

Comment (Swess) [reply src/HttpLib/HttpRequestMethod.java:30]: Would yield nulll

File: src/HttpLib/HttpRequestHandler.java (hunk 1 above)
Comment (Swess) [reply src/HttpLib/HttpRequestHandler.java]: It depends on how we want the handler to act/react to it. Should it handle everything (like right now) and return an empty response, or should it throw directly and let the application deal with a wrongly formatted response ?

**Review by Nwitz — APPROVED** [review]

**Pull request merged**